azkaban-common/src/main/java/azkaban/imagemgmt/rampup/ImageRampupManagerImpl.java

```diff
@@ -72,23 +78,69 @@ public ImageRampupManagerImpl(final ImageRampupDao imageRampupDao,
   }
 
   @Override
-  public Map<String, String> getVersionForAllImageTypes() throws ImageMgmtException {
+  public Map<String, VersionInfo> getVersionForAllImageTypes() throws ImageMgmtException {
     final Map<String, List<ImageRampup>> imageTypeRampups = this.imageRampupDao
         .getRampupForAllImageTypes();
     final List<ImageType> imageTypeList = this.imageTypeDao.getAllImageTypes();
     final Set<String> imageTypes = new TreeSet<>();
     for (final ImageType imageType : imageTypeList) {
       imageTypes.add(imageType.getName());
     }
-    return this.processAndGetVersionForImageTypes(imageTypes, imageTypeRampups);
+    final Set<String> remainingImageTypes = new TreeSet<>();
+    final Map<String, ImageVersionMetadata> imageTypeVersionMap = this
+        .processAndGetVersionForImageTypes(imageTypes, imageTypeRampups, remainingImageTypes);
+    // Throw exception if there are left over image types
+    if (!remainingImageTypes.isEmpty()) {
+      throw new ImageMgmtException("Could not fetch version for below image types. Reasons: "
+          + " 1. There is not active rampup plan in the image_rampup_plan table. 2. There is no "
+          + " acitve version in the image_versions table. Image Types: " + remainingImageTypes);
```

Comment:
Nit: fix typo `active` in this and other exception messages.

Reply:
Thanks. Done.